app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.service
+
+import cats.data.EitherT
+import cats.instances.future._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.http.Status._
+import uk.gov.hmrc.cgtpropertydisposals.connectors.TaxEnrolmentConnector
+import uk.gov.hmrc.cgtpropertydisposals.models.{Address, EnrolmentRequest, Error, KeyValuePair, SubscriptionDetails}
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentServiceImpl.TaxEnrolmentResponse
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentServiceImpl.TaxEnrolmentResponse.TaxEnrolmentCreated
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[TaxEnrolmentServiceImpl])
+trait TaxEnrolmentService {
+  def allocateEnrolmentToGroup(cgtReference: String, subscriptionDetails: SubscriptionDetails)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, TaxEnrolmentResponse]
+}
+
+object TaxEnrolmentServiceImpl {
+
+  sealed trait TaxEnrolmentResponse
```

Comment:
this would make more sense belonging in the companion object of the `TaxEnrolmentService` rather than it's implementation since the trait references this - it isn't implementation specific

Reply:
yes, refactored.